pkg/server/server.go

```diff
@@ -20,12 +18,17 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/metric"
 	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 )
 
 type ExperimentalFeatureFlag string
 
 const (
 	AuthorizationModelIDHeader = "openfga-authorization-model-id"
+
+	// XHttpCode is used for overriding the standard HTTP code
+	XHttpCode = "x-http-code"
```

Comment:
I think we should avoid having transport specific code in the server itself (other than grpc since the server implements the RPC service itself). Since grpc-gateway is essentially just a reverse-proxy middleware ontop of the RPC server, I think it'd be preferable to leave this in the `http` middleware package.